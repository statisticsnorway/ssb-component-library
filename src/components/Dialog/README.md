Dialog
========

> A component for showing messages to the user

### Usage

#### HTML

```html
<div class="ssb-dialog">
    <div class="icon-panel">
        <Icon />
    </div>
    <div class="dialog-content">
        <span class="dialog-title">A fitting title</span>
        <p>Insert content here</p>
    </div>
</div>
```

#### React

```jsx harmony
<Dialog type='warning' title="A fitting title">
  Insert content here
</Dialog> 
```

Available props:

| Name       | Type           | Description  |
| ---------- | ------------- | ----- |
| children   | Required node | Content within the wrapper |
| title | Required string | Bold text as a title |
| type | 'info' or 'warning | Changes color and icon |
