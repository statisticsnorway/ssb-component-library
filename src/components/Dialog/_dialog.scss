.ssb-dialog {
  @include open-sans;
  display: flex;
  max-width: 1180px;
  min-width: 100%;
  min-height: 95px;
  width: 100%;

  .icon-panel {
    color: $ssb-white;
    display: flex;
    flex-shrink: 0;
    font-size: 36px;
    justify-content: center;
    padding-top: 20px;
    width: 90px;

    .icon {
      font-size: 40px;
    }
  }

  .dialog-content {
    padding: 16px;

    .dialog-title {
      font-weight: bold;
    }

    .content {
      margin-top: 8px;
    }
  }

  &.warning {
    border: 2px solid $ssb-red-3;

    .icon-panel {
      background: $ssb-red-3;
    }
  }

  &.info {
    border: 2px solid $ssb-blue-3;

    .icon-panel {
      background: $ssb-blue-3;
    }
  }
}
