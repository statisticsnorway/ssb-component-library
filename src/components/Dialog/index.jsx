import React from 'react';
import PropTypes from 'prop-types';
import { AlertCircle, AlertTriangle } from 'react-feather';
import Paragraph from '../Paragraph/index.jsx';

const Dialog = ({children, title, type}) => (
	<div className={`ssb-dialog ${type}`}>
		<div className="icon-panel">
			{type === 'warning' && <AlertTriangle size="36" />}
			{type === 'info' && <AlertCircle size="36" />}
		</div>
		<div className="dialog-content">
			<span className="dialog-title">{title}</span>
			<Paragraph>{children}</Paragraph>
		</div>
	</div>
);

Dialog.defaultProps = {
	type: 'info',
};

Dialog.propTypes = {
	children: PropTypes.node.isRequired,
	title: PropTypes.string.isRequired,
	type: PropTypes.oneOf([
		'info',
		'warning',
	]),
};

export default Dialog;
