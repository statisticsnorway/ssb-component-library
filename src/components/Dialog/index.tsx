import React, { type ReactNode } from 'react'
import { AlertTriangle, Info } from 'react-feather'

interface DialogProps {
  children: ReactNode
  className?: string
  title: string
  type?: 'info' | 'warning'
}

const Dialog: React.FC<DialogProps> = ({ children, className, title, type = 'info' }) => (
  <div className={`ssb-dialog ${type}${className ? ` ${className}` : ''}`}>
    <div className='icon-panel'>
      {type === 'warning' && <AlertTriangle size={45} className='icon' />}
      {type === 'info' && <Info size={45} className='icon' />}
    </div>
    <div className='dialog-content'>
      <span className='dialog-title'>{title}</span>
      <div className='content'>{children}</div>
    </div>
  </div>
)

export default Dialog
