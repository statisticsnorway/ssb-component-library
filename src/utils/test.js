import fs from 'fs';
import path from 'path';
import { render as rtlRender } from '@testing-library/react';

export * from '@testing-library/react';
export * from '@testing-library/jest-dom';

const styles = fs.readFileSync(path.relative(process.cwd(), './lib/bundle.css'), 'utf8');

export function render(ui, options) {
	const view = rtlRender(ui, options);

	const styleElement = document.createElement('style');
	styleElement.innerHTML = styles;
	document.body.appendChild(styleElement);

	return view;
}
