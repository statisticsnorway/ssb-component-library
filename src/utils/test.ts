import fs from 'fs'
import path from 'path'
import { RenderOptions, RenderResult, render as rtlRender } from '@testing-library/react'
import { ReactNode } from 'react'

export * from '@testing-library/react'

const styles = fs.readFileSync(path.relative(process.cwd(), './lib/bundle.css'), 'utf8')

export function render(ui: ReactNode, options?: RenderOptions): RenderResult {
  const view = rtlRender(ui, options)

  const styleElement = document.createElement('style')
  styleElement.innerHTML = styles
  document.body.appendChild(styleElement)

  return view
}
