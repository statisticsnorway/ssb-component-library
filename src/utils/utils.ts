// utils.ts
const classNames = (...classes: (string | boolean | undefined)[]): string => classes.filter(Boolean).join(' ')

export default classNames
