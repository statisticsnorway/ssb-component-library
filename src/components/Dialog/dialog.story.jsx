import React from 'react';
import { storiesOf } from '@storybook/react';
import centered from '@storybook/addon-centered/react';
import Dialog from './index';

const placeholderText = `
Here you can write a informative text and inform the user of what has happened.
In the instance the informative warning is regarding moved statistics or diagrams,
please attach a link to where the data can be found. `;

storiesOf('Dialog', module).addDecorator(centered)
	.add('Info', () => (
		<div>
			<Dialog
				title="This article has been moved"
			>{placeholderText}
			</Dialog>
		</div>
	))
	.add('Warning', () => (
		<div>
			<Dialog
				title="Be aware of the dog"
				type="warning"
			>{placeholderText}
			</Dialog>
		</div>
	));
