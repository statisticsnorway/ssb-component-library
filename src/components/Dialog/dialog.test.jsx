import React from 'react'

import { render } from '../../utils/test'
import Dialog from './index'

describe('Dialog component', () => {
  test('Matches the snapshot', () => {
    const { asFragment } = render(<Dialog title='Dialog Title'>Dialog</Dialog>)
    expect(asFragment()).toMatchSnapshot()
  })
  test('Change dialog type Toggles classNames correctly', () => {
    const { asFragment } = render(
      <Dialog title='Dialog Title' type='warning'>
        Dialog
      </Dialog>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
