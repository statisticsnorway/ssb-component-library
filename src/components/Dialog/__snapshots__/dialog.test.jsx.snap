// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog component Change dialog type Toggles classNames correctly 1`] = `
<DocumentFragment>
  <div
    class="ssb-dialog warning"
  >
    <div
      class="icon-panel"
    >
      <svg
        class="icon"
        fill="none"
        height="45"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="45"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10.29 3.86L1.82 18a2 2 0 0 0 1.71 3h16.94a2 2 0 0 0 1.71-3L13.71 3.86a2 2 0 0 0-3.42 0z"
        />
        <line
          x1="12"
          x2="12"
          y1="9"
          y2="13"
        />
        <line
          x1="12"
          x2="12.01"
          y1="17"
          y2="17"
        />
      </svg>
    </div>
    <div
      class="dialog-content"
    >
      <span
        class="dialog-title"
      >
        Dialog Title
      </span>
      <div
        class="content"
      >
        Dialog
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Dialog component Matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="ssb-dialog info"
  >
    <div
      class="icon-panel"
    >
      <svg
        class="icon"
        fill="none"
        height="45"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="45"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          r="10"
        />
        <line
          x1="12"
          x2="12"
          y1="16"
          y2="12"
        />
        <line
          x1="12"
          x2="12.01"
          y1="8"
          y2="8"
        />
      </svg>
    </div>
    <div
      class="dialog-content"
    >
      <span
        class="dialog-title"
      >
        Dialog Title
      </span>
      <div
        class="content"
      >
        Dialog
      </div>
    </div>
  </div>
</DocumentFragment>
`;
